.github/workflows/docs-diff.yml

```diff
@@ -0,0 +1,240 @@
+# Check for changes in docs output.
+name: Docs Diff
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+jobs:
+  check-label:
+    name: Check for "Docs" label
+    runs-on: ubuntu-latest
+    outputs:
+      hadLabel: ${{ steps.label.outcome == 'skipped' }}
+    steps:
+      - id: label
+        name: Check for "Docs" label
+        run: exit 1 # Fail the job!
+        if: "!contains(github.event.pull_request.labels.*.name, 'Docs')"
+        continue-on-error: true
+  build-head:
+    name: Build docs on HEAD.
+    runs-on: ubuntu-latest
+    needs: check-label
+    if: ${{ needs.check-label.outputs.hadLabel == 'false' }}
+    steps:
+      - uses: actions/cache@v2
+        id: cache
+        with:
+          path: |
+            docs/public
+          key: docs-${{ github.sha }}
+      - uses: actions/checkout@v2
+        if: steps.cache.outputs.cache-hit != 'true'
```

Comment:
What are these?

Reply:
This caches the doc build so that we don't have to rebuild on every run. For example if you add the label after the first run of this, the docs will already be cached.

Reply:
Isn't the `actions/cache` steps doing the caching?

I'm specifically wondering about the `if: steps.cache.outputs.cache-hit != 'true'` multiple times. 

Is the idea that there will only be a cache hit if the job made it all the way to the end. Therefore a cache hit implies that the CI run was previously successful for the same files?
And we need to do the skipping because even with the docs build cached it still takes a little while to run?

Reply:
It still takes around 4 minutes to run even if the docs build is cached unless we skip.

Reply:
Okay. A little ugly, but seems like the best way to do that currently.

We're also going to need to keep this in sync with any other docs CI changes.